### PR TITLE
feat(sdk-api): Refactor API into separate package

### PR DESCRIPTION
**Pull request by bitgoAaron**
* Adds 2 new packages:
   * `sdk-core` : package for defining core level interfaces & common functions. 0 dependencies. 
   * `sdk-api`: package for making generic API calls to wallet-platform, handles HMAC, encryption/decryption etc. Very few dependencies.
* Updates entire repo to use [TS Project References](https://blog.logrocket.com/boost-your-productivity-with-typescript-project-references/).
   * Now you can run TypeScript compiler watches across the entire repo in realtime w/o having to stop & rebuild everything. 
   * `yarn dev` from the root of the repo, and start working cross-module with ease.  
   
 [Epic](https://bitgoinc.atlassian.net/browse/BG-41831)

### CODE REVIEW

File: modules/sdk-api/src/api.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export function setRequestQueryString(req: superagent.SuperAgentRequest): void {
  * Right now, it is very permissive with the timestamp variance.
  */
 export function verifyResponse(
-  bitgo: BitGo,
+  bitgo: BitGoAPI,
```
Comment (argjv) [inline modules/sdk-api/src/api.ts:179]: Should this be `walletAPI`? We will have multiple services with different clients in the near future

File: modules/sdk-api/src/encrypt.ts [hunk 2]
```diff
@@ -15,10 +15,11 @@ export function bytesToWord(bytes?: Uint8Array | number[]): number {
...
 }
 
-export function encrypt(password: string, plaintext: string, {
-  salt = randomBytes(8),
-  iv = randomBytes(16),
-} = {}): string {
+export function encrypt(
+  password: string,
+  plaintext: string,
+  { salt = randomBytes(8), iv = randomBytes(16) } = {}
```
Comment (argjv) [inline modules/sdk-api/src/encrypt.ts:21]: Why do we include these params? New type?

File: tsconfig.packages.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "files": [],
+  "references": [
+    {
+      "path": "./modules/account-lib"
+    },
+    {
+      "path": "./modules/bitgo"
+    },
+    {
+      "path": "./modules/blockapis"
```
Comment (argjv) [inline tsconfig.packages.json:11]: we probably need to rename this package to adhere to our naming convention, possibly `explorer-api`

**Comment by argjv** [comment]
LGTM but would like to get a second review.

File: modules/sdk-api/package.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "webpack-dev": "webpack",
+    "webpack-prod": "webpack --mode=production --node-env=production",
+    "prepublishOnly": "npm run webpack-prod"
+  },
+  "author": "BitGo Inc.",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/BitGo/BitGoJS.git",
+    "directory": "modules/statics"
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-api/package.json:25]: modules/sdk-api?

File: modules/sdk-core/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "lint": "eslint --quiet .",
+    "precommit": "lint-staged",
+    "prepare": "npm run build"
+  },
+  "author": "BitGo Inc.",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/BitGo/BitGoJS.git",
+    "directory": "modules/statics"
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/package.json:21]: modules/sdk-core?

File: modules/sdk-api/src/encrypt.ts (hunk 2 above)
Comment (bitgoAaron) [reply modules/sdk-api/src/encrypt.ts:21]: I'm not sure, but this isn't a change I made. I just `git mv` 'd this file. This is the same as it is currently:
https://github.com/BitGo/BitGoJS/blob/master/modules/bitgo/src/encrypt.ts#L21

File: modules/sdk-api/package.json (hunk 4 above)
Comment (bitgoAaron) [reply modules/sdk-api/package.json:25]: good catch. ACK

File: modules/sdk-api/package.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  "publishConfig": {
+    "access": "public"
+  },
+  "nyc": {
+    "extension": [
+      ".ts"
+    ]
+  },
+  "dependencies": {
+    "@bitgo/sdk-core": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (bitgoAaron) [reply modules/sdk-api/package.json:47]: not a circular dependency. 
`bitgo` -> `@bitgo/sdk-api` -> `@bitgo/sdk-core`

sdk-core has no dependencies.

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Review by argjv — APPROVED** [review]

**Pull request merged**